src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/DefineAccessorsForAttributeArguments.cs

```diff
@@ -123,9 +128,9 @@ private static IEnumerable<IParameterSymbol> GetAllPublicConstructorParameters(I
             var propertiesMap = new Dictionary<string, IPropertySymbol>(StringComparer.OrdinalIgnoreCase);
             foreach (INamedTypeSymbol currentType in attributeType.GetBaseTypesAndThis())
             {
-                foreach (IPropertySymbol property in currentType.GetMembers().Where(m => m.Kind == SymbolKind.Property))
+                foreach (var member in currentType.GetMembers())
```

Comment:
I would rather prefer this being written as:
```csharp
foreach (var property in currentType.GetMembers().OfType<IPropertySymbol())
```

Reply:
That's just one more enumerator that is immediately iterated and only once.